UI-UX-Developer.md

```diff
@@ -0,0 +1,77 @@
+
+**Objective**: Display the climate risk data in a table format, allowing users to sort and filter the dataset.
+
+### Tasks:
+1. Create a data table component.
+2. Load and display the sample dataset with a given year selection (from Problem 1) in the table.
+3. Implement sorting functionality on reasonable columns.
+4. Implement filter functionality on reasonable columns, especially risk factors.
+
+## Problem 3: Visualize Risk Over Time with Line Graphs
+
+**Objective**: Create a line graph to visualize the changes in risk levels over time for a given location, Asset or Business Category.
+
+### Tasks:
+1. Set up a charting library (e.g., Chart.js, D3.js, Highcharts).
+2. Implement a line graph component that displays the Risk Rating over time (Year) for a selected location (Lat, Long), Asset, or Business Category.
+3. Add interactivity to the graph, such as tooltips displaying Asset Name, Risk Rating, Risk Factors, and Year.
+4. Implement controls for selecting different locations, Assets, or Business Categories to visualize their risk levels over time.
+    - You may need to perform some data aggregation in order to achieve this.
+
+## Problem 4: Integrate Components and Optimize Performance
+
+**Objective**: Combine the components from the previous problems into a cohesive web app, ensuring optimal performance and user experience.
+
+Tasks:
+1. Design a user interface that integrates the map, data table, and line graph components.
+2. Implement state management to handle user interactions and data flow between components (e.g., selecting a location on the map updates the line graph and data table).
+3. Optimize the app's performance by implementing lazy loading for components and efficient data handling, such as pagination for the data table.
+
+## Notes
+
+- You are encouraged to use any open source libraries to assist with the given problems.
+- You can use AI assistants (such as ChatGPT or GitHub Copilot), but make sure to include the full interaction/chat log along with your solution.
+- Attribute tutorials, Stackoverflow Q&As, and any other references along with your solution.
+- You are encouraged to ask any questions you may have during the hiring process, including but not limited to questions related to the work sample itself.
+- Keep in mind that both the amount and the dimension of the data we work with are much higher in reality, so try to demonstrate that your solutions can handle beyond the sample scale.
+- Try to think of yourself as the user of the web app, who may want to have certain questions answered, such as:
+    - What are the top risk factors by ranking?
+    - What regions are at the most risk?
+    - What risk factors impact a particular category the most?
+    - Any others that you can think of...
+
+## Submission
+
+- Host your solution code in a open source repository (e.g. GitHub).
+- Deploy your solutions through a suitable service (see below for recommended services).
```

Comment:
I think this part is kind of a duplication of `Host your solution code in a open source repository`, right? or maybe i just confusing of what this statement try to tell me what to do!

Reply:
The first one is for us to review the source code, the second is to see in deployed/action form (so we can use and assess the web app without installing on our machines)

Reply:
ok, if that is the case... then the deployment should be a (bonus point) or (good to have) because i don't think i have been asked to do that ever since i have been interviewed for a full stack position. What do you think? @ChrisHampu.

Reply:
For reference, it's a balance to simplify our review work, and minimize risk of potentially malicious code to run from our work station.

Reply:
Yeah, maybe add some recommendation might help the candidate a little bit.